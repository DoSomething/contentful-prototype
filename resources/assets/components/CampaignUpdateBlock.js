import React from 'react';
import { markdown } from '../helpers';

import Block from './Block/index';
import Figure from './Figure';

const CampaignUpdateBlock = (props) => {
  const { title, content } = props.fields;

  const author = 'Lizzy Divine';
  const jobTitle = 'Campaign Lead';
  const avatar = 'http://images.contentful.com/e5mq1t8pfsum/6bZ26cuCC4YmAgYA6YasQs/2ce7b11d25384edc35fcfe7c24002bce/Pic_1_PP_DITL_EDivine.jpeg';

  return (
    <Block>
      <h2>{title}</h2>
      <div dangerouslySetInnerHTML={markdown(content)} />
      <Figure image={avatar}>
        <h3>{author}</h3>
        <p className="footnote">{jobTitle}</p>
      </Figure>
    </Block>
  );
};

export default CampaignUpdateBlock;
