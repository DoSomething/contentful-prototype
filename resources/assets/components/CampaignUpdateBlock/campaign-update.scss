@import "~@dosomething/forge/scss/toolkit";

.campaign-update__content {
  margin-bottom: $base-spacing;
}

.campaign-update__content.-tweet p {
  font-size: $font-large;
}

.campaign-update__content.-tweet p img {
  float: left;
  padding-right: $base-spacing/2;
}
