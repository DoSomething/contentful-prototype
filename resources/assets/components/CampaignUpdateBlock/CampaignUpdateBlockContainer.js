/* global window */

import { connect } from 'react-redux';
import { find } from 'lodash';
import CampaignUpdateBlock from './CampaignUpdateBlock';
import { makeShareLink } from '../../helpers';

const mapStateToProps = (state, props) => {
  const campaignUpdateFields = props.fields ||
    find(state.campaign.activityFeed, { id: props.id }).fields;

  return {
    fields: campaignUpdateFields,
    shareLink: makeShareLink('campaigns', {
      domain: window.location.origin,
      slug: state.campaign.slug,
      key: props.id,
    }),
  };
};

export default connect(mapStateToProps)(CampaignUpdateBlock);
