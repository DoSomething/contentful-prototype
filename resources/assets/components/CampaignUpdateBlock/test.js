import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CampaignUpdateBlock from './index';

test('Campaign Update Block with no additional content snapshot test', () => {
  const component = shallow(
    <CampaignUpdateBlock
      fields={{
        title: 'Heyo!',
        content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit non mi porta gravida at eget metus. Vestibulum id ligula porta felis euismod semper. Vestibulum id ligula porta felis euismod semper. Morbi leo risus, porta ac consectetur ac, vestibulum at eros.',
      }}
    />,
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Campaign Update Block with additional content snapshot test', () => {
  const component = shallow(
    <CampaignUpdateBlock
      fields={{
        title: 'Heyo!',
        content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit non mi porta gravida at eget metus. Vestibulum id ligula porta felis euismod semper. Vestibulum id ligula porta felis euismod semper. Morbi leo risus, porta ac consectetur ac, vestibulum at eros.',
        additionalContent: {
          author: 'Braümhilda Snosages',
        },
      }}
    />,
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Campaign Update Block with additional content as tweet snapshot test', () => {
  const component = shallow(
    <CampaignUpdateBlock
      fields={{
        title: 'Heyo!',
        content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit non mi porta gravida at eget metus vestibulum.',
        additionalContent: {
          author: 'Braümhilda Snosages',
        },
      }}
    />,
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
