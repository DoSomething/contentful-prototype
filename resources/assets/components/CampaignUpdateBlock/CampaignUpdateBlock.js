import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Markdown from '../Markdown';
import { BlockWrapper } from '../Block';
import Embed from '../Embed';
import Byline from '../Byline';
import { ShareContainer } from '../Share';
import './campaign-update.scss';

const CampaignUpdateBlock = (props) => {
  const { title, content = 'Let\'s Do This!', link, additionalContent = null } = props.fields;

  const isTweet = content.length < 144;

  const share = (
    <ShareContainer
      link={props.shareLink}
      variant="icon"
      parentSource="campaign-update"
    />
  );

  return (
    <BlockWrapper title="Campaign Update" id={props.id}>
      { isTweet ? null : <h2>{title}</h2> }

      <Markdown className={classnames('campaign-update__content', { '-tweet': isTweet })}>
        {content}
      </Markdown>

      { link ? <Embed url={link} /> : null }

      { additionalContent ?
        <Byline
          author={additionalContent.author}
          jobTitle={additionalContent.jobTitle}
          avatar={additionalContent.avatar}
          share={share}
        />
        : null }
    </BlockWrapper>
  );
};

CampaignUpdateBlock.propTypes = {
  id: PropTypes.string.isRequired,
  shareLink: PropTypes.string.isRequired,
  fields: PropTypes.shape({
    title: PropTypes.string,
    content: PropTypes.string,
    additionalContent: PropTypes.shape({
      author: PropTypes.string,
      jobTitle: PropTypes.string,
      avatar: PropTypes.string,
    }),
  }).isRequired,
};

export default CampaignUpdateBlock;
