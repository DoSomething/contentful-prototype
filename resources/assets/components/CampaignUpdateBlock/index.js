import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import Markdown from '../Markdown';
import Block, { BlockTitle } from '../Block';
import { Figure } from '../Figure';
import Embed from '../Embed';
import DEFAULT_AVATAR from './default-avatar.png';
import './campaign-update.scss';

const Byline = ({ author, jobTitle, avatar }) => (
  <Figure size="small" alignment="left" verticalAlignment="center" image={avatar} alt={`picture of ${author}`} imageClassName="avatar">
    <strong>{author}</strong><br />
    <p className="footnote">{jobTitle}</p>
  </Figure>
);

Byline.propTypes = {
  author: PropTypes.string.isRequired,
  jobTitle: PropTypes.string,
  avatar: PropTypes.string,
};

Byline.defaultProps = {
  jobTitle: 'DoSomething.org Staff',
  avatar: DEFAULT_AVATAR,
};

const CampaignUpdateBlock = (props) => {
  const { title, content, link, additionalContent } = props.fields;
  const { author, jobTitle, avatar } = additionalContent;

  const isTweet = content.length < 144;

  return (
    <Block>
      <BlockTitle>Campaign Update</BlockTitle>
      { isTweet ? null : <h2>{title}</h2> }
      <Markdown className={classnames('campaign-update__content', { '-tweet': isTweet })}>
        {content}
      </Markdown>
      { link ? <Embed url={link} /> : null }
      { author ? <Byline author={author} jobTitle={jobTitle} avatar={avatar} /> : null}
    </Block>
  );
};

CampaignUpdateBlock.propTypes = {
  fields: PropTypes.shape({
    title: PropTypes.string,
    content: PropTypes.string,
    additionalContent: PropTypes.shape({
      author: PropTypes.string.isRequired,
      jobTitle: PropTypes.string,
      avatar: PropTypes.string,
    }),
  }).isRequired,
};

export default CampaignUpdateBlock;
