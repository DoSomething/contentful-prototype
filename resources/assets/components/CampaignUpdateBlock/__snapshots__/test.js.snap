// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign Update Block with additional content as tweet snapshot test 1`] = `
<BlockWrapper
  id="1234512345"
  title="Campaign Update"
>
  <Markdown
    className="campaign-update__content -tweet"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit non mi porta gravida at eget metus vestibulum.
  </Markdown>
  <Byline
    author="Braümhilda Snosages"
    avatar={Object {}}
    jobTitle="DoSomething.org Staff"
  />
</BlockWrapper>
`;

exports[`Campaign Update Block with additional content snapshot test 1`] = `
<BlockWrapper
  id="1234512345"
  title="Campaign Update"
>
  <h2>
    Heyo!
  </h2>
  <Markdown
    className="campaign-update__content"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit non mi porta gravida at eget metus. Vestibulum id ligula porta felis euismod semper. Vestibulum id ligula porta felis euismod semper. Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
  </Markdown>
  <Byline
    author="Braümhilda Snosages"
    avatar={Object {}}
    jobTitle="DoSomething.org Staff"
  />
</BlockWrapper>
`;

exports[`Campaign Update Block with no additional content snapshot test 1`] = `
<BlockWrapper
  id="1234512345"
  title="Campaign Update"
>
  <h2>
    Heyo!
  </h2>
  <Markdown
    className="campaign-update__content"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id elit non mi porta gravida at eget metus. Vestibulum id ligula porta felis euismod semper. Vestibulum id ligula porta felis euismod semper. Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
  </Markdown>
</BlockWrapper>
`;
